### PR TITLE
DCS-493 HDC - Don't log confidential headers

### PR DESCRIPTION
**Pull request by bell-pa**

### CODE REVIEW

File: server/data/restClientBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+export const constantTokenSource = (token): TokenSource => async () => {
+  if (!token) {
+    throw Error('Unauthorised access')
+  }
+  return token
+}
+
+export const dynamicTokenSource = (signInService, oauthServiceName: string): TokenSource => async () => {
```
Comment (andrewrlee) [inline server/data/restClientBuilder.ts:36]: could possibly call this something about client credentials / anonymous / system to system token?

File: server/data/restClientBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+          .get(`${apiUrl}${path}`)
+          .query(query)
+          .agent(keepaliveAgent)
+          .set('Authorization', `Bearer ${token}`)
+          .set(headers)
+          .retry(2, (err, res) => {
+            if (res) {
+              if (res.status >= 300 && res.status !== NOT_FOUND && res.status !== UNAUTHORIZED) {
+                logger.warn(
+                  `Retry handler found API error with status: '${res.status}', message: '${res.error?.message}'`
```
Comment (andrewrlee) [inline server/data/restClientBuilder.ts:70]: probably don't want to retry on any 4XX client error?

File: server/data/nomisClientBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    getUserInfo(userName) {
+      return oauthRestClient.getResource(`/api/user/${userName}`)
+    },
+
+    getLoggedInUserInfo(): Promise<Profile> {
+      return oauthRestClient.getResource(`/api/user/me`)
+    },
+
+    getUserRoles(): Promise<Role> {
+      return oauthRestClient.getResource(`/api/user/me/roles`)
```
Comment (andrewrlee) [inline server/data/nomisClientBuilder.ts:179]: Out of scope for this story but is there anything to stop us from splitting out the auth calls into a separate client?

File: server/data/restClientBuilder.ts (hunk 1 above)
Comment (bell-pa) [reply server/data/restClientBuilder.ts]: Done

File: server/data/restClientBuilder.ts (hunk 2 above)
Comment (bell-pa) [reply server/data/restClientBuilder.ts]: Yes, of course! No idea why I did this.  Now retrying on res.status >= 500.

**Pull request merged**